plugins/rollbar_dashboards/app/controllers/rollbar_dashboards/dashboards_controller.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class RollbarDashboards::DashboardsController < ApplicationController
+  MAX_RQL_JOB_WAIT_TIME = 5
+
+  def deploy_dashboard
+    deploy = Deploy.find(params.require(:deploy_id))
+
+    data = deploy.project.rollbar_dashboards_settings.map do |setting|
+      items = Rails.cache.fetch(['deploy', setting], expires_in: 10.minutes) do
+        client = RollbarDashboards::Client.new(setting)
+
+        rql_job_id = client.create_rql_job(deploy_rql_query(deploy))
+        rql_job_id ? wait_for_rql_job_results(rql_job_id, client) : nil
+      end
+
+      [setting, items&.first(RollbarDashboards::Setting::NUM_ITEMS)]
+    end
+
+    render(
+      partial: 'rollbar_dashboards/dashboard',
+      collection: data,
+      as: :dashboard_data,
+      locals: { title: "Top #{RollbarDashboards::Setting::NUM_ITEMS} Items That Occurred This Deploy" }
+    )
+  end
+
+  private
+
+  def wait_for_rql_job_results(job_id, client)
+    MAX_RQL_JOB_WAIT_TIME.times do
+      result = client.rql_job_result(job_id)
+      return result if result.nil? || result != RollbarDashboards::Client::RQL_JOB_WAITING_TERM
+      sleep(1)
```

Comment:
it's kinda strange to sleep and then stop the loop :D 
`times do |i| ... break if i >= (MAX_RQL_JOB_WAIT_TIME - 1)`